src/Server/WishList.fs

```diff
@@ -1,52 +1,47 @@
 /// Wish list API web parts and data access functions.
 module ServerCode.WishList
 
+open System.Threading.Tasks
+open Microsoft.AspNetCore.Http
 open Giraffe
-open RequestErrors
-open ServerErrors
 open ServerCode.Domain
-open Microsoft.AspNetCore.Http
-open Microsoft.Extensions.Logging
-open System.Threading.Tasks
+open ServerTypes
 
 /// Handle the GET on /api/wishlist
-let getWishList (getWishListFromDB: string -> Task<WishList>) next (ctx: HttpContext) =
-    Auth.useToken next ctx (fun token -> task {
-        try
+let getWishList (getWishListFromDB : string -> Task<WishList>) (token : UserRights) : HttpHandler =
+     fun (next : HttpFunc) (ctx : HttpContext) ->
+        task {
             let! wishList = getWishListFromDB token.UserName
             return! ctx.WriteJsonAsync wishList
-        with exn ->
-            let msg = "Database not available"
-            let logger = ctx.GetLogger "wishlist"
-            logger.LogError (EventId(), exn, msg)
-            return! SERVICE_UNAVAILABLE "Database not available" next ctx
-    })
+        }
+
+let private invalidWishList =
+    RequestErrors.BAD_REQUEST "WishList is not valid"
+
+let inline private forbiddenWishList username =
+    sprintf "WishList is not matching user %s" username
+    |> RequestErrors.FORBIDDEN
 
 /// Handle the POST on /api/wishlist
-let postWishList (saveWishListToDB: WishList -> Task<unit>) next (ctx: HttpContext) =
-    Auth.useToken next ctx (fun token -> task {
-        try
+let postWishList (saveWishListToDB: WishList -> Task<unit>) (token : UserRights) : HttpHandler =
+    fun (next : HttpFunc) (ctx : HttpContext) ->
+        task {
             let! wishList = ctx.BindJsonAsync<Domain.WishList>()
 
-            if token.UserName <> wishList.UserName then
-                return! FORBIDDEN (sprintf "WishList is not matching user %s" token.UserName) next ctx
-            else
-                if Validation.verifyWishList wishList then
+            match token.UserName.Equals wishList.UserName with
```

Comment:
using Equals instead of = is usally considered unidiomatic in F#

Reply:
Ok cool... didn't know that either... I think this is still something I'm used from C#. Should probably read some F# coding standard guidelines somewhere to make myself more familiar with this stuff!